UnitsNet.Tests/QuantityTypeConverterTest.cs

```diff
@@ -0,0 +1,348 @@
+
+            Assert.Throws<ArgumentException>(() => unitsNETTypeConverter.ConvertFrom(context, culture, "1m^2"));
+        }
+
+        [Theory]
+        [InlineData(typeof(Length))]
+        [InlineData(typeof(IQuantity))]
+        [InlineData(typeof(object))]
+        public void ConvertTo_GiveWrongType_ThrowException(Type value)
+        {
+            UnitsNetTypeConverter<Length> unitsNETTypeConverter = new UnitsNetTypeConverter<Length>();
+            ITypeDescriptorContext context = new TypeDescriptorContext("SomeMemberName", new Attribute[] { });
+            Length length = Length.FromMeters(1);
+
+            Assert.Throws<NotSupportedException>(() => unitsNETTypeConverter.ConvertTo(length, value));
+        }
+
+        [Theory]
+        [InlineData(typeof(string))]
+        public void ConvertTo_GiveRigthType_DoConvertion(Type value)
+        {
+            UnitsNetTypeConverter<Length> unitsNETTypeConverter = new UnitsNetTypeConverter<Length>();
+            ITypeDescriptorContext context = new TypeDescriptorContext("SomeMemberName", new Attribute[] { });
+            Length length = Length.FromMeters(1);
+
+            Assert.Equal("1 m", unitsNETTypeConverter.ConvertTo(length, value));
+        }
+
+        [Fact]
+        public void ConvertTo_GiveSomeValues_ConvertToString()
+        {
+            UnitsNetTypeConverter<Length> unitsNETTypeConverter = new UnitsNetTypeConverter<Length>();
+            ITypeDescriptorContext context = new TypeDescriptorContext("SomeMemberName", new Attribute[] { });
+            Length length = Length.FromMeters(1);
+
+            Assert.Equal("1 m", unitsNETTypeConverter.ConvertTo(length, typeof(string)));
+            Assert.Equal("1 m", unitsNETTypeConverter.ConvertTo(context, culture, length, typeof(string)));
+
+            context = new TypeDescriptorContext("SomeMemberName", new Attribute[]
+            {
+                new DefaultUnitAttribute(Units.LengthUnit.Centimeter),
+                new ConvertToUnitAttribute(Units.LengthUnit.Millimeter)
+            });
+
+            Assert.Equal("1 m", unitsNETTypeConverter.ConvertTo(length, typeof(string)));
+            Assert.Equal("1 m", unitsNETTypeConverter.ConvertTo(context, culture, length, typeof(string)));
+
+            context = new TypeDescriptorContext("SomeMemberName", new Attribute[]
+            {
+                new DefaultUnitAttribute(Units.LengthUnit.Centimeter),
+                new ConvertToUnitAttribute(Units.LengthUnit.Millimeter),
+                new DisplayAsUnitAttribute(Units.LengthUnit.Decimeter)
+            });
+
+            Assert.Equal("1 m", unitsNETTypeConverter.ConvertTo(length, typeof(string)));
+            Assert.Equal("10 dm", unitsNETTypeConverter.ConvertTo(context, culture, length, typeof(string)));
+        }
+
+        [Fact]
+        public void ConvertTo_TestDisplayAsFormating_ConvertToFormatedString()
```

Comment:
Typo `Formatting` and `Formatted`
Maybe rename to suffix `_ReturnsQuantityStringWithDisplayUnit`.